tfhe/src/core_crypto/commons/mod.rs

```diff
@@ -143,49 +145,77 @@ pub mod test_tools {
         }
     }
 
-    pub fn assert_noise_distribution<First, Second, Element>(
-        first: &First,
-        second: &Second,
-        dist: impl DispersionParameter,
-    ) where
-        First: Container<Element = Element>,
-        Second: Container<Element = Element>,
-        Element: UnsignedTorus,
-    {
-        use rand_distr::Distribution;
-
-        let std_dev = dist.get_standard_dev();
-        let confidence = 0.95;
-        let n_slots = first.container_len();
-
-        // allocate 2 slices: one for the error samples obtained, the second for fresh samples
-        // according to the std_dev computed
-        let mut sdk_samples = vec![0.0_f64; n_slots];
-
-        // recover the errors from each ciphertexts
-        sdk_samples
-            .iter_mut()
-            .zip(first.as_ref().iter().zip(second.as_ref().iter()))
-            .for_each(|(out, (&lhs, &rhs))| *out = torus_modular_distance(lhs, rhs));
-
-        // fill the theoretical sample vector according to std_dev using the rand crate
-        let mut theoretical_samples: Vec<f64> = Vec::with_capacity(n_slots);
-        let normal = rand_distr::Normal::new(0.0, std_dev).unwrap();
-        for _i in 0..n_slots {
-            theoretical_samples.push(normal.sample(&mut rand::thread_rng()));
-        }
-
-        // compute the kolmogorov smirnov test
-        let result = kolmogorov_smirnov::test_f64(
-            sdk_samples.as_slice(),
-            theoretical_samples.as_slice(),
-            confidence,
-        );
+    /// Based on Shapiro-Francia normality test
+    pub fn normality_test_f64(samples: &[f64], alpha: f64) -> (f64, f64, bool) {
```

Comment:
I might be worth here to create a struct type with clear member name as the return type instead of returning a tuple.

to make lines like `normality_test_f64(&samples, 0.05).2` a bit more explicit

Reply:
agreed, will do